### PR TITLE
Create a task

### PR DESCRIPTION
**Pull request by mjzac**

### CODE REVIEW

File: handling-advanced-ui-events/create-new-task/README.md [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+      return state;
+  }
+};
+```
+
+Now we have our reducer ready.
+
+Next, we need to add the actual API call that needs to be invoked to create a task.
+
+Let's open `src/context/task/actions.ts`
```
Comment (reenas) [inline handling-advanced-ui-events/create-new-task/README.md:121]:
```suggestion
Let's open `src/context/task/actions.ts` and update it as per the following code.
```

**Comment by reenas** [comment]
@mjzac A general comment about this lesson, there are many code snippets included in this. Can we add more inline explanation in them to students understand what is happening?

File: handling-advanced-ui-events/create-new-task/README.md [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+      return state;
+  }
+};
+```
+
+Now we have our reducer ready.
+
+Next, we need to add the actual API call that needs to be invoked to create a task.
+
+Let's open `src/context/task/actions.ts` and update it as per the following code.
```
Comment (reenas) [inline handling-advanced-ui-events/create-new-task/README.md:121]: @mjzac Do you want to add some inline comments where needed against important sections here?

File: handling-advanced-ui-events/create-new-task/README.md [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+    <TasksProvider>
+      <ProjectDetails />
+      <Outlet />
+    </TasksProvider>
+  );
+};
+```
+
+Now, we have to create the modal window component.
+Let's create a folder named `tasks` in `src/pages`. Inside this folder, let's create a file named `NewTask.tsx` with following content. We will use `react-hook-form` to manage the form.
```
Comment (reenas) [inline handling-advanced-ui-events/create-new-task/README.md:227]: @mjzac Do you want to add some inline comments where needed against important sections here?

File: handling-advanced-ui-events/create-new-task/README.md (hunk 2 above)
Comment (mjzac) [reply handling-advanced-ui-events/create-new-task/README.md:127]: I have added inline comments for better clarity

**Pull request merged**